modules/oe_theme_helper/src/Plugin/field_group/FieldGroupFormatter/FieldInPageNavigationFormatter.php

```diff
@@ -0,0 +1,140 @@
+      if (!is_array($element[$value]) || $this->excludeField($value, $element[$value], $processed_object['#fieldgroups']) === TRUE) {
+        continue;
+      }
+      // We will add anchors when the element, field or group, has a label.
+      $label = $this->fieldGetLabel($value, $processed_object);
+      // Add menu element and anchor.
+      if ($label !== '') {
+        // If the group has a default id, use it as anchor link.
+        if (isset($processed_object['#fieldgroups'][$value]) && isset($processed_object['#fieldgroups'][$value]->format_settings['id']) && $processed_object['#fieldgroups'][$value]->format_settings['id'] !== '') {
+          $id = $processed_object['#fieldgroups'][$value]->format_settings['id'];
+        }
+        else {
+          $id = "inline-nav-" . $i++;
+        }
+        $links[] = [
+          'href' => "#" . $id,
+          'label' => $label,
+        ];
+        $element[$value]['#id'] = $id;
+        $element[$value]['#attributes']['id'] = $id;
+      }
+    }
+    $element['#links'] = $links;
+    if (count($links) > 0) {
+      $element['#inline_title'] = t('Page contents');
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender(&$element, $rendering_object) {
+    parent::preRender($element, $rendering_object);
+    $this->process($element, $rendering_object);
+  }
+
+  /**
+   * Check if the field should create an in page navigation entry.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $field
+   *   The field to check.
+   * @param array $fieldgroups
+   *   List of groups.
+   *
+   * @return bool
+   *   Include or not the field.
+   */
+  private function excludeField(string $field_name, array $field, array $fieldgroups) :bool {
+    // Exclude empty fields.
+    if (!in_array($field_name, array_keys($fieldgroups)) && !isset($field['#items'])) {
+      return TRUE;
+    }
+    // Exclude empty groups.
+    if (in_array($field_name, array_keys($fieldgroups)) && count($field) < 1) {
+      return TRUE;
+    }
+    // Exclude groups with only empty fields.
+    if (in_array($field_name, array_keys($fieldgroups))) {
+      foreach ($field as $value) {
+        // If there is a field with values, include the group.
+        if (isset($value['#items'])) {
+          return FALSE;
+        }
+      }
+      return TRUE;
+    }
+    return FALSE;
+  }
+
+  /**
+   * Get the label used on the field.
+   *
+   * @param string $field_name
+   *   Name of the field.
+   * @param array $processed_object
+   *   Entity processed.
+   *
+   * @return string
+   *   Label of the field.
+   */
+  private function fieldGetLabel(string $field_name, array $processed_object): string {
+    $label = '';
+    if (isset($processed_object['#fieldgroups'][$field_name]->label)) {
+      if ($processed_object['#fieldgroups'][$field_name]->label !== '') {
+        $label = $processed_object['#fieldgroups'][$field_name]->label;
```

Comment:
can we make sure this returns a string

Reply:
Changed to ensure we always get a string